addons/web/static/src/legacy/js/fields/basic_fields.js

```diff
@@ -576,6 +650,53 @@ var FieldChar = InputField.extend(TranslatableFieldMixin, {
     // Private
     //--------------------------------------------------------------------------
 
+
+    /**
+     * @override
+     */
+    init: function () {
+        this._super(...arguments);
+        if (this.nodeOptions && this.nodeOptions.dynamic_placeholder) {
+            // When the dynamic placeholder is active, the recordData
+            // need to be updated when `mailing_model_real` change
+            this.resetOnAnyFieldChange = true;
+        }
+    },
+
+    /**
+     * Open a Model Field Selector in order to select fields
+     * and create a dynamic placeholder string with or without
+     * a default text value.
+     *
+     * @public
+     * @param {String} baseModel
+     * @param {Array} chain
+     *
+     */
+    openDynamicPlaceholder: async function (baseModel, chain = []) {
+        const modelSelector = await this._openNewModelSelector(baseModel, chain);
+        if (this.$el.first().parents('div.modal').length === 0) {
+            modelSelector.$el.css('margin-top', this.$el.height() + 12);
```

Comment:
This looks hackish in FieldChar. What is the purpose of this line ?

Reply:
It's to be sure it is position properly when the Dynamic placeholder (DP) is open Inside a bootstrap modal.

Since the modal is relative, we need to position the DP with the `div.modal` as a reference. 
hence the hackish calculation.